docker/segment_processor/process_segment.py

```diff
@@ -4,26 +4,65 @@
 import argparse
 import numpy as np
 import pandas as pd
-from vqasynth.datasets.segment import CLIPSeg, SAM, sample_points_from_heatmap
+from vqasynth.datasets.segment import CLIPSeg, SAM2, sample_points_from_heatmap
+from transformers import AutoProcessor, AutoModelForCausalLM 
+import torch
 
 
-clipseg = CLIPSeg(model_name="CIDAS/clipseg-rd64-refined")
-sam = SAM(model_name="facebook/sam-vit-huge", device="cuda")
+
+device = "cuda" 
+assert torch.cuda.is_available()
+torch_dtype = torch.float16
+
+model = AutoModelForCausalLM.from_pretrained("microsoft/Florence-2-large", torch_dtype=torch_dtype, trust_remote_code=True).to(device)
+processor = AutoProcessor.from_pretrained("microsoft/Florence-2-large", trust_remote_code=True)
+sam = SAM2(model_name="facebook/sam2-hiera-large", device="cuda")
+
+def florence_caption(image):
+    prompt = "<MORE_DETAILED_CAPTION>"
+    inputs = processor(text=prompt, images=image, return_tensors="pt").to(device, torch_dtype)
+
+    generated_ids = model.generate(
+        input_ids=inputs["input_ids"],
+        pixel_values=inputs["pixel_values"],
+        max_new_tokens=1024,
+        num_beams=3,
+        do_sample=False
+    )
+    generated_text = processor.batch_decode(generated_ids, skip_special_tokens=False)[0]
+    parsed_answer = processor.post_process_generation(generated_text, task=prompt, image_size=(image.width, image.height))
+
+    prompt = parsed_answer[prompt]
+
+    inputs = processor(text=prompt, images=image, return_tensors="pt").to(device, torch_dtype)
+
+    generated_ids = model.generate(
+        input_ids=inputs["input_ids"],
+        pixel_values=inputs["pixel_values"],
+        max_new_tokens=1024,
+        num_beams=3,
+        do_sample=False
+    )
+    generated_text = processor.batch_decode(generated_ids, skip_special_tokens=False)[0]
+    parsed_answer = processor.post_process_generation(generated_text, task="<CAPTION_TO_PHRASE_GROUNDING>", image_size=(image.width, image.height))
+    
+    parsed_answer = parsed_answer['<CAPTION_TO_PHRASE_GROUNDING>']
+    bboxes = parsed_answer['bboxes']
+    labels = parsed_answer['labels']
+    return list(zip(bboxes, labels))
+
 
 def segment_image_data(row):
     try:
-        preds = clipseg.run_inference(row["image"], row["captions"])
 
-        sampled_points = []
+        caption_points = florence_caption(row["image"])
```

Comment:
@harshbhatt7585 this returns bounding boxes so we may need to use another method of SAM2 accepting bbox prompts

Reply:
yes, I yet have to test this, I will fix this